### PR TITLE
feat: test case coverage for verify

### PR DESCRIPTION
**Pull request by gjj**
For v3 verify

- added tests for unaltered data, altered data/target hash/proofs/merkle root
- added batched document samples (so that we can test altered proofs/merkle root)

Also solved the following issue

> currently `verify` only check the salts and obfuscated parts, so adding new data anywhere in the doc (except for salts and obfuscated) will go undetected and the doc remains valid, which is not expected

with the following check

- count(visible) + count(obfuscated) should match count(decodedSalts) in verify.ts

### CODE REVIEW

File: test/e2e.v3.test.ts [hunk 1]
```diff
@@ -167,9 +167,11 @@ describe("v3 E2E Test Scenarios", () => {
       const newDocument = await wrapDocument(datum[2], { version: SchemaId.v3 });
       const obfuscatedDocument = await obfuscate(newDocument, ["key2"]);
 
-      const verified = verifySignature(obfuscatedDocument);
-      expect(verified).toBe(true);
-      expect(validateSchema(obfuscatedDocument)).toBe(true);
+      const signatureValidity = verifySignature(obfuscatedDocument);
```
Comment (Nebulis) [inline test/e2e.v3.test.ts:170]: I find it useless to create a temporary var for this, can directly call the function in the expect

File: src/v3/verify.test.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      const verifiableCredential = {
+        ...cloneDeep(sampleVerifiableCredential),
+        recipient: {
+          fakename: "Recipient Name"
+        }
+      };
+      const verified = verify(verifiableCredential);
+      expect(verified).toBe(false);
+    });
+    test("returns false for documents with added data", () => {
```
Comment (Nebulis) [inline src/v3/verify.test.ts:42]: returns false for document with removed data ?

File: src/v3/verify.test.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+              effectiveDate: "2020-06-05T00:00:00Z"
+            }
+          ]
+        }
+      };
+      const verified = verify(verifiableCredential);
+      expect(verified).toBe(false);
+    });
+
+    test("returns true for documents with unaltered data (with proofs)", () => {
```
Comment (Nebulis) [inline src/v3/verify.test.ts:67]: for all the (with proofs test), I would suggest to create a new describe block that will contain all of them

File: src/v3/verify.test.ts (hunk 2 above)
Comment (gjj) [reply src/v3/verify.test.ts:42]: fixed, changed to `returns false for documents with additional data not part of salt`

File: src/v3/verify.test.ts (hunk 3 above)
Comment (gjj) [reply src/v3/verify.test.ts:67]: updated!

File: test/e2e.v3.test.ts (hunk 1 above)
Comment (gjj) [reply test/e2e.v3.test.ts]: fixed!

**Pull request merged**